hydra-node/test/Hydra/Events/FileBasedSpec.hs

```diff
@@ -0,0 +1,76 @@
+-- | Tests for the 'EventSource' and 'EventSink' implementation in 'Hydra.Events.FileBased'.
+module Hydra.Events.FileBasedSpec where
+
+import Hydra.Prelude hiding (label)
+import Test.Hydra.Prelude
+
+import Hydra.Events (EventSink (..), EventSource (..), StateEvent (..), getEvents, putEvent)
+import Hydra.Events.FileBased (eventPairFromPersistenceIncremental)
+import Hydra.Ledger.Simple (SimpleTx)
+import Hydra.Persistence (PersistenceIncremental (..), createPersistenceIncremental)
+import Test.QuickCheck (forAllShrink, ioProperty, sublistOf, (===))
+import Test.QuickCheck.Gen (listOf)
+
+spec :: Spec
+spec = do
+  describe "eventPairFromPersistenceIncremental" $ do
+    prop "can handle continuous events" $
+      forAllShrink genContinuousEvents shrink $ \events ->
+        ioProperty $ do
+          withEventSourceAndSink $ \EventSource{getEvents} EventSink{putEvent} -> do
+            forM_ events putEvent
+            loadedEvents <- getEvents
+            pure $
+              loadedEvents === events
+
+    prop "can handle non-continuous events" $
+      forAllShrink (sublistOf =<< genContinuousEvents) shrink $ \events ->
```

Comment:
forgive my ignorance but, does this expression: `sublistOf =<< genContinuousEvents`
really means non-continuous? 

for me it looks like a sparse list of continuos events; i mean getting a `sublistOf` of a continuous list seems to hold the same property by definition, am i wrong?

Reply:
https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/src/Test.QuickCheck.Gen.html#sublistOf

looks like `sublistOf` just flips a coin to include or not include each element, so it can generate discontinouous sublists; this matches the convention that math takes where a "subsequence" is just ordered elements selected from a sequence, rather than a range of contiguous elements.